### PR TITLE
Adding more `wasm-parser` builds

### PR DESCRIPTION
**Pull request by Saeid-Za**
I've added `bundler` and `node` target builds and added code samples to showcase the usage.

Related Issue: [Bug: Can not use @oxc-parser/wasm without error](https://github.com/oxc-project/oxc/issues/2828)

### CODE REVIEW

**Comment by Saeid-Za** [comment]
We could further decrease the package size, by shipping only one wasm file, which i think could be shared between builds

**Comment by Boshen** [comment]
This is amazing! I'll take a look at this tomorrow and tinker with the release.

**Comment by Saeid-Za** [comment]
Is there anything I could help with?

**Comment by Boshen** [comment]
Sorry it'll take a while to go through a huge PR like this, and get to this when I get the time.

**Comment by Saeid-Za** [comment]
Oh, sure, please take your time
If there is anything I could help with, don't hesitate to mention me !

File: wasm/parser/scripts/build.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+const outDir = resolve(join(__dirname, "../../../npm/wasm-parser"))
+
+async function main() {
+    consola.start("Building WASM target ...");
+
+    await rm(outDir, { recursive: true, force: true })
+    await mkdir(outDir, { recursive: true })
+
+    await Promise.all([
+        buildProject("bundler", join(outDir, "bundler")),
```
Comment (Boshen) [inline wasm/parser/scripts/build.ts:18]: Why is this changed from `web` to `bundler`?

**Comment by Boshen** [comment]
Closed by #2916

Released v0.1.0 https://www.npmjs.com/package/@oxc-parser/wasm

**Comment by Saeid-Za** [comment]
Regarding why build targets `bundler` instead of `web`, most users use a bundler like vite or webpack as their primary tooling in front-end development.
The bundler target, supports both these tools. (Hence the vue/vite demo in the PR), and the `wasm-pack` itself, defaults to it because the majority use these toolings.
The web target, requires manual initiation and I think I couldn't make it work with vite (don't remember why though).
